### ISSUE TITLE
Confused about neural adapter and the dataset in the paper

### ISSUE THREAD

**Issue by Helicqin** [liah-chan/transferNER#1]
Thanks for your time.

I have read your great paper, but I am a little confused about the neural adapter and the dataset.
In my opinion, the source model and target model should input the same sentence with different label sequence in the transfer learning process, but the dataset seems opposite.

> For the purpose of our experiment, we divide the CONLL train set in 80%/20% as DS and DT

I would appreciate it if you could make some explanations about it.

**Comment by liah-chan** [comment]
Hi! The setting in this paper is that, in the initial step, we have model `M_S` trained on source data `D_S` with `n` NE categories. Then in the subsequent step, we do not have access to `D_S` anymore, only the trained model `M_S`. In the subsequent step, we want the target model to be able to recognize the NEs in `D_T` (including original `n` NE categories, plus `m` NE categories that were never seen in `D_S`).
This is to mimic the industrial scenario of using NER since often there are privacy concerns of releasing the training data.

It is natural in transfer learning that the target data is different from the source data, hence the separation of the dataset as described in the paper. Can you elaborate on why you think the source model and target model should input the same sentence with different label? In my opinion, in this way, the model is presented with redundant data (at least in our experiment settings in the paper), since the only difference would be the labels regarding the unseen NE category.

In terms of the neural adapter, it is used for providing an 'additional input' to the target model. This additional input comes from passing the target data also as an input to the source model, and get the representation of the prediction from source model. Intuitively, you can think of this as enhancing the prediction of the target model. 

Hope this clarify some of your confusions!

**Comment by Helicqin** [comment]
Thanks for your detailed reply! I have understood your meaning.

![neural adapter](https://user-images.githubusercontent.com/21177644/56138109-c1bd0900-5fc8-11e9-94a7-758204f42e77.PNG)

The adapter is intended to solve the problem of disagreement in annotations between the same source and target sequences. I seriously think of your detailed explanations. So the target data is used as an input to the source model, which both input sequences of the source model and target model are of the same length. That is why I misunderstood it. 

 Sorry for your precious time.  Would you mind me asking another question? The setting in the paper is kind of easy, which just took care of increasing new entities. However, we may meet more complicated situations where entity labeling standard changes. For example, entity `A` and `B` change to the new entity `C`, the opposite, etc.  How the adapter could make a difference to these complex situations remains worth exploring.

Anyway, much grateful for your generous and precious reply!

**Comment by liah-chan** [comment]
You are welcome! I am glad to help:)
That is a very interesting scenario and indeed worth exploring. I think it comes down to if the adapter can help generalize to situations of NE category changes other than addition, such as replacement, merging or the refinement of NE categories. Some of these scenarios are in essence still facing the labeling disagreement problem in the source data and the target data, so technically adapter would be able to help. There are some other issues to concern too. For example, is more target data needed for the target model to reach comparable performance, or if we should have higher learning rate in the transfer step etc.
